cpp/tests/structs/utilities_tests.cpp

```diff
@@ -305,9 +305,9 @@ TYPED_TEST(TypedStructUtilitiesTest, StructOfStructWithNullsAtAllLevels)
                                     expected_nums_col_3,
                                     expected_strings_col}};
 
-  CUDF_TEST_EXPECT_TABLES_EQUAL(expected,
-                                cudf::structs::detail::flatten_nested_columns(
-                                  table, {}, {}, cudf::structs::detail::column_nullability::FORCE));
+  auto flattened_table = cudf::structs::detail::flatten_nested_columns(
+    table, {}, {}, cudf::structs::detail::column_nullability::FORCE);
+  CUDF_TEST_EXPECT_TABLES_EQUAL(expected, flattened_table->flattened_columns());
```

Comment:
Another comment that's not about the changes in this PR :D
The lines where `flattened_columns` is used read a bit poorly. `flattened_columns` to me implies that a vector of columns is returned. IMO `view` would be a better name, since the function returns a `table_view` and a call site like `flattened_table->view()` reads exactly right.
This is just a side note, not requesting any changes.

Reply:
Oh thanks.
I would like to change the entire name of the class `flattened_table` instead, as it is not a table but a data structure holding a table_view and many more things. So this should probably be addressed in a future PR.

Reply:
I agree that this should be a separate PR